sdcm/kafka/kafka_producer.py

```diff
@@ -0,0 +1,410 @@
+
+        return value
+
+
+class KafkaProducerThread(KafkaThreadBase):
+    """
+    Thread that produces records to a Kafka topic.
+    Supports 'avro' and 'json-with-schema' record formats.
+    """
+
+    def __init__(self, tester, params: SCTConfiguration, timeout: int, **kwargs):
+        super().__init__(tester, params, timeout, **kwargs)
+        self.produced_count = 0
+        self.failed_count = 0
+        self.current_key = 1
+
+        self.record_size = int(self.kafka_options.get('record-size', 1024))
+
+        producer_properties = {}
+        if 'producer-property' in self.kafka_options:
+            if isinstance(self.kafka_options['producer-property'], dict):
+                producer_properties.update(self.kafka_options['producer-property'])
+            elif isinstance(self.kafka_options['producer-property'], list):
+                for prop in self.kafka_options['producer-property']:
+                    producer_properties.update(prop)
+        if 'property' in self.kafka_options:
+            if isinstance(self.kafka_options['property'], dict):
+                producer_properties.update(self.kafka_options['property'])
+            elif isinstance(self.kafka_options['property'], list):
+                for prop in self.kafka_options['property']:
+                    producer_properties.update(prop)
+
+        producer_config = {
+            'bootstrap.servers': ','.join(self.kafka_addresses),
+        }
+        # The following parameters can be used to tune/optimize performance and reliability of the producer
+        # producer_config['linger.ms'] = int(producer_properties.get('linger.ms', kwargs.get('linger_ms', 5)))
+        # producer_config['batch.num.messages'] = int(
+        #     producer_properties.get('batch.num.messages', kwargs.get('batch_num_messages', 10000)))
+        # producer_config['acks'] = producer_properties.get('acks', kwargs.get('acks', '1'))
+        # producer_config['compression.type'] = producer_properties.get(
+        #     'compression.type', kwargs.get('compression_type', 'none'))
+
+        producer_config.update(producer_properties)
+
+        self.producer = Producer(producer_config)
+        self.wait_for_topic(self.topic, timeout=60)
+
+    def get_topics(self) -> list[str]:
+        admin_client = Producer({'bootstrap.servers': ','.join(self.kafka_addresses)})
+        topics = admin_client.list_topics(timeout=10).topics.keys()
+        LOGGER.debug("Available topics: %s", topics)
+        return topics
+
+    def wait_for_topic(self, topic: str, timeout: int) -> None:
+        wait_for(lambda: topic in self.get_topics(), text=f"waiting for topic={topic}", timeout=timeout)
+
+    def delivery_report(self, err: str, record: Message) -> None:
+        if err:
+            self.failed_count += 1
+            LOGGER.error("Kafka record delivery failed: %s", err)
+        else:
+            self.produced_count += 1
```

Comment:
`self.produced_count` -> `self.passed_count`

Reply:
For me `passed` in a name is immediately something about passing the tests. I doubt it fits here.
Producer/produced is a term from Kafka itself, for that purpose it was selected here. Another alternative could be delivered_count ('delivery' term is also used in kafka extensively).

Reply:
yes, then 'delivered_count' and 'undelivered_count' should be used in that case (I got confused with `failed_count`)

Reply:
Right, `failed` can also be considered as some test status related thing at the first place )
Changed the names.